# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build gcc-g++-build tinyxml2-build vulkan-headers-build"

PKG_CONFIG_PATH=/wl/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
rm -rf tinyxml2
rm -rf Vulkan-Headers
PKG_CONFIG_PATH=/wl/lib/pkgconfig cmake -DVULKAN_HPP_INSTALL=ON -DVULKAN_HPP_RUN_GENERATOR=ON -DCMAKE_INSTALL_INCLUDEDIR=/wl/include -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/wl/include/vulkan
install -m 644 vulkan/vulkan.hpp $DESTDIR/wl/include/vulkan

# build.pc
install -d $DESTDIR/wl/share/pkgconfig
cat > $DESTDIR/wl/share/pkgconfig/vulkan-hpp-build.pc << EOF
Name: Vulkan-Hpp
Version: 20201104
Description: Vulkan C++ API Header
Requires: $REQUIRES

devel=\\
/wl/include/vulkan/vulkan.hpp
EOF

exit
--- Vulkan-Hpp-20201104.orig/CMakeLists.txt
+++ Vulkan-Hpp-20201104/CMakeLists.txt
@@ -54,6 +54,12 @@
 
 message("CMAKE_CXX_STANDARD = <${CMAKE_CXX_STANDARD}>")
 
+find_package(PkgConfig QUIET)
+pkg_check_modules(VulkanRegistry QUIET vulkan-registry)
+if (VulkanRegistry_FOUND)
+  pkg_get_variable(VulkanRegistry_DIR vulkan-registry pkgdatadir)
+endif()
+
 if (NOT DEFINED VulkanRegistry_DIR)
   if (DEFINED VULKAN_HPP_VULKAN_HEADERS_SRC_DIR)
     set(VulkanRegistry_DIR "${VULKAN_HPP_VULKAN_HEADERS_SRC_DIR}/registry")
@@ -85,17 +91,19 @@
   VulkanHppGenerator.cpp
 )
 
-if (NOT DEFINED VULKAN_HPP_TINYXML2_SRC_DIR)
-  set(VULKAN_HPP_TINYXML2_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tinyxml2")
-endif()
-
-set(TINYXML2_SOURCES
-  ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.cpp
-)
+if (IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tinyxml2)
+  if (NOT DEFINED VULKAN_HPP_TINYXML2_SRC_DIR)
+    set(VULKAN_HPP_TINYXML2_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tinyxml2")
+  endif()
 
-set(TINYXML2_HEADERS
-  ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.h
-)
+  set(TINYXML2_SOURCES
+    ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.cpp
+  )
+
+  set(TINYXML2_HEADERS
+    ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.h
+  )
+endif()
 
 source_group(headers FILES ${HEADERS})
 source_group(sources FILES ${SOURCES})
@@ -118,7 +126,11 @@
   target_compile_options(VulkanHppGenerator PRIVATE -Wall -Wextra -pedantic -Werror)
 endif(MSVC)
 
-target_include_directories(VulkanHppGenerator PRIVATE ${VULKAN_HPP_TINYXML2_SRC_DIR})
+if (DEFINED VULKAN_HPP_TINYXML2_SRC_DIR)
+  target_include_directories(VulkanHppGenerator PRIVATE ${VULKAN_HPP_TINYXML2_SRC_DIR})
+else()
+  target_link_libraries(VulkanHppGenerator tinyxml2)
+endif()
 
 option (VULKAN_HPP_RUN_GENERATOR "Run the HPP generator" OFF)
 if (VULKAN_HPP_RUN_GENERATOR)
